security/pkg/server/ca/server.go

```diff
@@ -289,13 +294,19 @@ func (s *Server) applyServerCertificate() (*tls.Certificate, error) {
 	return &cert, nil
 }
 
+// authenticate goes through a list of authenticator (provided client cert, k8s jwt, and ID token)
+// and authenticates if one of them is valid.
 func (s *Server) authenticate(ctx context.Context) *authenticate.Caller {
 	// TODO: apply different authenticators in specific order / according to configuration.
 	var errMsg string
 	for id, authn := range s.authenticators {
+		// If SDS is not enabled, authenticate will not perform k8s jwt authentication.
+		if !s.sdsEnabled && authn.AuthenticatorType() == authenticate.KubeJWTAuthenticatorType {
```

Comment:
Can we check sdsEnabled in line 214 instead? We don't put the Kube JWT authenticator in the authenticator list if sdsEnabled is true.

Reply:
That's better. Thanks!